packages/x-charts/src/context/SeriesContextProvider.tsx

```diff
@@ -42,13 +48,9 @@ if (process.env.NODE_ENV !== 'production') {
   SeriesContext.displayName = 'SeriesContext';
 }
 
-const seriesTypeFormatter: {
-  [type in ChartSeriesType]?: (series: any, dataset?: DatasetType) => any;
-} = {
-  bar: barSeriesFormatter,
-  scatter: scatterSeriesFormatter,
-  line: lineSeriesFormatter,
-  pie: pieSeriesFormatter,
+export type SeriesFormatterConfig<T extends ChartSeriesType = ChartSeriesType> = {
+  // TODO replace the function type by Formatter<K>
```

Comment:
Is this todo meant to be in?

Reply:
Not in this PR, because it impacts such a big part of the codebase that it would require a dedicated PR